pkg/ingress/apisix_route.go

```diff
@@ -292,6 +292,26 @@ func (c *apisixRouteController) handleSyncErr(obj interface{}, errOrigin error)
 	c.workqueue.AddRateLimited(obj)
 }
 
+func (c *apisixRouteController) isApisixRouteEffective(ar kube.ApisixRoute) bool {
+	var class string
+
+	if ar.GroupVersion() == kube.ApisixRouteV1 {
+		class = ar.V1().Spec.IngressClassName
+	} else if ar.GroupVersion() == kube.ApisixRouteV2alpha1 {
+		class = ar.V2alpha1().Spec.IngressClassName
+	} else if ar.GroupVersion() == kube.ApisixRouteV2beta1 {
+		class = ar.V2beta1().Spec.IngressClassName
+	} else {
+		class = c.controller.cfg.Kubernetes.IngressClass
```

Comment:
Actually the else won't be entered, once the logic here was executed, then there are some programming faults occurred, we may just:

1. Panic, or
2. log the error and just return false.